modules/ocl/cv_surview_fisheye_dewarp.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * cv_surview_fisheye_dewarp.h - dewarp fisheye image for surround view
+ *
+ *  Copyright (c) 2016-2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: Junkai Wu <junkai.wu@intel.com>
+ */
+
+#ifndef XCAM_CV_SURVIEW_FISHEYE_DEWARP_H
+#define XCAM_CV_SURVIEW_FISHEYE_DEWARP_H
+
+#include <base/xcam_common.h>
+#include <base/xcam_buffer.h>
+#include <dma_video_buffer.h>
+#include <smartptr.h>
+#include "xcam_obj_debug.h"
+#include "image_file_handle.h"
+#include "cv_base_class.h"
+#include "cv_calibration_parser.h"
+
+#include <ocl/cl_context.h>
+#include <ocl/cl_device.h>
+#include <ocl/cl_memory.h>
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/core/ocl.hpp>
+
+namespace XCam {
+
+class BowlDataConfig
+{
+
+public:
+    explicit BowlDataConfig();  
+
+    int image_height;
+    int image_heightZ0;
+
+    float a, b, c;
+    float angle_start, angle_end;
+
+};
+
+class CVSurViewFisheyeDewarp : public CVBaseClass
+{
+
+public:
+    explicit CVSurViewFisheyeDewarp ();
+
+    void FisheyeDewarp(float *mapTable, IntrinsicParameter intrinsic_param, ExtrinsicParameter extrinsic_param, BowlDataConfig bowl_config);

```

Comment:
1. all class or structures as parameters should use &-reference. and if for readonly, use const XXX &;
2. move extrinsic/intrinsic params into member variables.
3. use pointer as less as possible: float *map_table => std::vector<float> &table.

for this case. change should be like:

     class FisheyeXXXX {
            typedef std::vector<float> MapTable;
            void fisheye_dewarp(MapTable &table,  uint32_t table_w, uint32_t table_h, uint32_t image_w, uint32_t image_h, const BowlDataConfig &bowl_config);
            void set_extrinsic_param (const ExtrinsicParameter  &param);
            void set_intrinsic_param (const IntrinsicParameter  &param);
            ...
      };

if table_size and image_size only used in fisheye_dewarp function. it's not necessary to have
set_map_table_size and set_image_size.